src/main/java/handler/WeatherHandler.java

```diff
@@ -0,0 +1,211 @@
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+
+
+public class WeatherHandler extends Handler{
+    private static final String TOKENS_DIRECTORY_PATH = "tokens/";
+    private static final String WEATHER_CENTER_FILE_PATH = "weather_center_data.csv";
+    private static final String LOCATION_NAME_URL_PATH = "location_name_url.csv";
+    private static String PW_url;
+    private static String currentUrl;
+    private static String key;
+    private ArrayList<Map<Object, Object>> predict_weather;
+    private Map<Object, Object> current_weather;
+    private final String location;
+
+    public WeatherHandler(){
+        ifOutput = true;
+        this.location = "基隆市";   //預設
+        weatherInit();
+    }
+    public WeatherHandler(String location){
+        ifOutput = true;
+        this.location = location;   
+        weatherInit();
+    }
+    public Map<Object, Object> getCurrentWeather(JSONObject json){
+        Map<Object, Object> weather= new HashMap<Object, Object>();
+        try{
+            
+            String temp= new String();
+            String HUMD= new String();; //相對濕度
+
+            JSONArray allLocation = json.getJSONObject("records").getJSONArray("location");
+            for(int i=0; i<allLocation.length(); i++){
+                JSONObject J= allLocation.getJSONObject(i);
+                String city = J.getJSONArray("parameter").getJSONObject(0).get("parameterValue").toString();
+                if(city.equals(location)){
+                    
+                    JSONArray weatherElement= J.getJSONArray("weatherElement");
+                    temp= weatherElement.getJSONObject(3).get("elementValue").toString();
+                    HUMD= weatherElement.getJSONObject(4).get("elementValue").toString();
+                    break;
+                }
+            }
+            weather.put("溫度（攝氏）", temp);
+            weather.put("相對濕度", HUMD);
+        }catch(Exception e){
+            System.err.println(e+" in getCurrentWeather");
+        }
+        return weather;
+    }
+    public ArrayList<Map<Object, Object>> getPredictWeather(){
+        return predict_weather;
+    }
+
+    public String[] openCsvFile(String path){
+        String data= new String();
+        try{
+            InputStream is = new FileInputStream(path);
+            BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+            String line = buf.readLine();
+            StringBuilder sb = new StringBuilder();
+            while (line != null) {
+                sb.append(line).append(",");
+                line = buf.readLine();
+            }
+            buf.close();
+            data= sb.toString();
+            return data.split(",");
+        }catch(IOException e){
+            System.out.println(e+" =>The File may not exist");
```

Comment:
Don't print any information in handler, you should use return String instead it.

Reply:
Use ```System.out.println()``` may influence the output layout such like if we want to run all handlers in parallel.